planck-cljs/test/planck/io_test.cljs

```diff
@@ -27,6 +27,29 @@
     (is (= js/Date (type (:modified (planck.io/file-attributes "/tmp")))))
     (is (number? (:file-size (planck.io/file-attributes "/tmp"))))))
 
+(deftest predicates-test
+  (testing "predicates"
+    (is (= false (planck.io/directory? nil)))
+    (is (= false (planck.io/exists? nil)))
+    (is (= false (planck.io/regular-file? nil)))
+    (is (= false (planck.io/symbolic-link? nil)))
+    (is (= false (planck.io/directory? "bogus")))
+    (is (= false (planck.io/exists? "bogus")))
+    (is (= false (planck.io/regular-file? "bogus")))
+    (is (= false (planck.io/symbolic-link? "bogus")))
+    (is (= true  (planck.io/directory? "/tmp")))
+    (is (= true  (planck.io/exists? "/tmp")))
+    (is (= false (planck.io/regular-file? "/tmp")))
+;    (is (= false (planck.io/symbolic-link? "/tmp")))   ; /tmp is a symbolic link on some systems (e.g. macos), but not others (e.g. most Linuxes), so it's not a great test case
+    (is (= false (planck.io/directory? "/dev/stdout")))
+    (is (= true  (planck.io/exists? "/dev/stdout")))
+    (is (= false (planck.io/regular-file? "/dev/stdout")))
+    (is (= true  (planck.io/symbolic-link? "/dev/stdout")))
+    (is (= false (planck.io/directory? "/bin/sh")))
```

Comment:
@mfikes I'm not at all confident that `/bin/sh` is a good example of a regular file that exists on virtually all Unixes.  Do you know if there's a better test case here?

Reply:
In answer to my own question, it looks like it's not a good choice - [on TravisCI (at least) `/bin/sh` is a shambolic link](https://travis-ci.org/planck-repl/planck/jobs/494643610#L682-L690).

Reply:
Yeah, to test a symbolic link, you could perhaps use `planck.shell` to create one.

Reply:
Implemented this in rev 1c16b0c - managed to used `planck.io` for all of it, rather than needing to bring in another ns.